gina-alaska/viirs-snow-metrics
# Download data from NSIDC DAAC

To test this PR:

Clone this repository on Elephant and do the following:

```sh
export INPUT_DIR=/export/datadir/$USER/viirs_inputs
export SCRATCH_DIR=$HOME/VIIRS_snow_metrics/scratch
export OUTPUT_DIR=/$HOME/VIIRS_snow_metrics/outputs
export DEV_MODE=True
```
 - Sign up for an Earthdata account if you do not already have one, and note your user name and password.
 - Read instructions for `download.py` in the README
 - Run `python config.py` to observe DEV_MODE behavior and to verify that directory config is working as expected on Elephant.
 - Run `python download.py`

Observe behavior when `download.py` runs with an empty `$INPUT_DIR`.
Observe behavior when `download.py` runs with `$INPUT_DIR` already contains data.
Observe behavior when invalid API credentials are provided.

Examine the log file `download.log`
Examine the contents of `$INPUT_DIR` after the download is complete: `ls -lhrt $INPUT_DIR`

Test with a 3 day download window (edit `luts.py`)
Test with a 3 month download window (edit `luts.py`)
Test with downloading the full time series (`export DEV_MODE=False`)
